deeprankcore/trainer.py

```diff
@@ -236,66 +236,45 @@ def _init_from_dataset(self, dataset: Union[GraphDataset, GridDataset]):
-                        raise ValueError(f"training dataset has node_features {dataset_train.node_features} while "
-                                         f"{dataset_other_name} dataset has node_features {dataset_other.node_features}")
-
-                    if dataset_other.edge_features != dataset_train.edge_features:
-                        raise ValueError(f"training dataset has edge_features {dataset_train.edge_features} while "
-                                         f"{dataset_other_name} dataset has edge_features {dataset_other.edge_features}")
-
-                    if dataset_other.clustering_method != dataset_other.clustering_method:
-                        raise ValueError(f"training dataset has clustering method {dataset_train.clustering_method} while "
-                                         f"{dataset_other_name} dataset has clustering method {dataset_other.clustering_method}")
-
-                elif isinstance(dataset_train, GridDataset) and isinstance(dataset_other, GridDataset):
-
-                    if dataset_other.features != dataset_train.features:
-                        raise ValueError(f"training dataset has features {dataset_train.features} while "
-                                         f"{dataset_other_name} dataset has features {dataset_other.features}")
-
-                else:
-                    raise TypeError(f"Training and {dataset_other_name} datasets are not the same type.\n"
-                                     "Make sure to use only graph or only grid datasets")
+        else:
+            # Make sure train dataset has valid type
+            if not isinstance(dataset_train, GraphDataset) and not isinstance(dataset_train, GridDataset):
+                raise TypeError(f"""test dataset is not the right type {type(dataset_train)}
+                                Make sure it's either GraphDataset or GridDataset""") 
+            
+            if dataset_val is not None:
+                # Check train parameter in valid is set as False.
+                if dataset_val.train is not False:
+                    raise ValueError(f"""valid dataset has train parameter {dataset_val.train}
+                                Make sure to set it as False""")
+                # Check dataset_train parameter in valid is equivalent to train which passed to Trainer.
+                if dataset_val.dataset_train != dataset_train:
+                    raise ValueError("valid dataset has different dataset_train parameter compared to the one given in Trainer.\n" +
+                                "Make sure to assign equivalent dataset_train in Trainer")
+                    
+            if dataset_test is not None:
+                # Check train parameter in test is set as False.
+                if dataset_test.train is not False:
+                    raise ValueError(f"""test dataset has train parameter {dataset_test.train}
+                                Make sure to set it as False""")
```

Comment:
The issue would be about standardization, which is the reason why train and dataset_train have been created in the first place. In the beginning, they were meant for retaining and using means and stds from the training set in the case of testing/validation sets creation. I think that such distinction should happen when the sets are defined, which is when we instantiate the different GraphDataset objects. 

I still have to finish the new review of this PR, but what I think could be the least redundant solution can be the following: create sets and use train/dataset_train parameters in the DeeprankDataset children, and then in the Trainer only verify that what is supposed to be train, valid, test has the right settings to be so. Otherwise signal to the user that the datasets configurations are wrong

Reply:
I see the point, and think you are right. I also think that that is the way it is set up now, so then my comments can be disregarded.